src/plugins/session-key/BaseSessionKeyPlugin.sol

```diff
@@ -0,0 +1,232 @@
+import {BasePlugin} from "../BasePlugin.sol";
+import {ISessionKeyPlugin} from "./interfaces/ISessionKeyPlugin.sol";
+import {ISingleOwnerPlugin} from "../owner/ISingleOwnerPlugin.sol";
+import {SingleOwnerPlugin} from "../owner/SingleOwnerPlugin.sol";
+
+/// @title Base Session Key Plugin
+/// @author Decipher ERC-6900 Team
+/// @notice This plugin allows some designated EOA or smart contract to temporarily
+/// own a modular account.
+/// This base session key plugin acts as a 'parent plugin' for all specific session
+/// keys. Using dependency, this plugin can be thought as a parent contract that stores
+/// session key duration information, and validation functions for session keys. All
+/// logics for session keys will be implemented in child plugins.
+/// It allows for session key owners to access MSCA both through user operation and
+/// runtime, with its own validation functions.
+/// Also, it has a dependency on SingleOwnerPlugin, to make sure that only the owner of
+/// the MSCA can add or remove session keys.
+
+contract BaseSessionKeyPlugin is BasePlugin, ISessionKeyPlugin {
+    using ECDSA for bytes32;
+
+    string public constant NAME = "Base Session Key Plugin";
+    string public constant VERSION = "1.0.0";
+    string public constant AUTHOR = "Decipher ERC-6900 Team";
+
+    mapping(address => mapping(address => mapping(bytes4 => bytes))) internal _sessionInfo;
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Execution functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc ISessionKeyPlugin
+    function addTemporaryOwner(address tempOwner, bytes4 allowedSelector, uint48 _after, uint48 _until) external {
+        if (_until <= _after) {
+            revert WrongTimeRangeForSession();
+        }
+        bytes memory sessionInfo = abi.encodePacked(_after, _until);
+        _sessionInfo[msg.sender][tempOwner][allowedSelector] = sessionInfo;
+        emit TemporaryOwnerAdded(msg.sender, tempOwner, allowedSelector, _after, _until);
+    }
+
+    /// @inheritdoc ISessionKeyPlugin
+    function removeTemporaryOwner(address tempOwner, bytes4 allowedSelector) external {
+        delete _sessionInfo[msg.sender][tempOwner][allowedSelector];
+        emit TemporaryOwnerRemoved(msg.sender, tempOwner, allowedSelector);
+    }
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Plugin view functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc ISessionKeyPlugin
+    function getSessionDuration(address account, address tempOwner, bytes4 allowedSelector)
+        external
+        view
+        returns (uint48 _after, uint48 _until)
+    {
+        (_after, _until) = _decode(_sessionInfo[account][tempOwner][allowedSelector]);
+    }
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Plugin interface functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc BasePlugin
+    function onInstall(bytes calldata data) external override {}
```

Comment:
I agree, that will add a convenient way to add session keys to the accounts. 

But I was thinking about adding it on the `onInstall` at child plugins(in this case TokenSessionKeyPlugin.sol`), since each session keys will be associated with them. I'll think more about it and update soon.